allennlp/modules/seq2vec_encoders/cls_pooler.py

```diff
@@ -36,6 +41,12 @@ def get_output_dim(self) -> int:
 
     @overrides
     def forward(self, tokens: torch.Tensor, mask: torch.Tensor = None):
-        # tokens is assumed to have shape (batch_size, sequence_length, embedding_dim).  We just
-        # want the first token for each instance in the batch.
-        return tokens[:, 0]
+        # tokens is assumed to have shape (batch_size, sequence_length, embedding_dim).
+        # mask is assumed to have shape (batch_size, sequence_length) with all 1s preceding all 0s.
+        if not self._cls_is_last_token:
+            return tokens[:, 0, :]
+        else:  # [CLS] at the end
+            if mask is None:
+                raise ValueError("Must provide mask for transformer models with [CLS] at the end.")
+            lengths = mask.sum(1)
+            return tokens[torch.arange(tokens.size(0)), lengths - 1, :]
```

Comment:
I'd recommend just calling this function: https://github.com/allenai/allennlp/blob/f4284dca4932d5623e36c5ffe3462104a370fe7d/allennlp/nn/util.py#L179.